src/OrleansTestingHost/TestingSiloHost.cs

```diff
@@ -403,12 +401,13 @@ public SiloHandle RestartSilo(SiloHandle instance)
                 {
                     Primary = newInstance;
                 }
-                else if (type == Silo.SiloType.Secondary)
+                else if (Secondary == instance)
```

Comment:
I recommend also changing the primary silo the same way.
Pretty sure this bug was introduced recently, as it used to be `Secondary == instance` before. and it was also Primary==instance